packages/gotrue/lib/src/constants.dart

```diff
@@ -27,6 +27,17 @@ enum AuthChangeEvent {
   mfaChallengeVerified,
 }
 
+extension AuthChangeEventExtended on AuthChangeEvent {
+  static AuthChangeEvent fromString(String? val) {
+    for (final event in AuthChangeEvent.values) {
+      if (event.name == val) {
+        return event;
+      }
+    }
+    return AuthChangeEvent.signedIn;
```

Comment:
I think it's strange to return `signedIn` as default case. Instead, I would simply return `null`. I know this doesn't really matter at the moment, because that method is only used once. But for future uses I think it's easier to understand to return `null` and let the method caller decide what's the default case should be.

Reply:
Sounds good. Updated!